src/main/scala/com/mwz/sonar/scala/scalastyle/ScalastyleRulesRepository.scala

```diff
@@ -0,0 +1,146 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+package com.mwz.sonar.scala
+package scalastyle
+
+import cats.instances.string._
+import cats.syntax.option._
+import org.scalastyle._
+import org.sonar.api.batch.rule.Severity
+import org.sonar.api.rule.RuleStatus
+import org.sonar.api.rules.RuleType
+import org.sonar.api.server.rule.{RuleParamType, RulesDefinition}
+
+/**
+ * Defines a repository for the Scalastyle inspections.
+ */
+final class ScalastyleRulesRepository extends RulesDefinition {
+  import ScalastyleRulesRepository._
+
+  override def define(context: RulesDefinition.Context): Unit = {
+    // Create an empty repository.
+    val repository = context
+      .createRepository(RepositoryKey, Scala.LanguageKey)
+      .setName(RepositoryName)
+
+    // Register each Scalastyle inspection as a repository rule.
+    ScalastyleInspections.AllInspections.foreach { inspection =>
+      val rule = repository.createRule(inspection.clazz)
+      rule.setInternalKey(inspection.clazz)
+      rule.setName(inspection.label)
+      rule.setMarkdownDescription(formatDescription(inspection))
+      rule.setActivatedByDefault(true) // scalastyle:ignore LiteralArguments
+      rule.setStatus(RuleStatus.READY)
+      rule.setSeverity(levelToSeverity(inspection.defaultLevel).name)
+      rule.setType(RuleType.CODE_SMELL)
+
+      // Create parameters.
+      inspection.params.foreach { param =>
+        rule
+          .createParam(param.name)
+          .setType(parameterTypeToRuleParamType(param.typ))
+          .setDescription(s"${param.label}: ${param.description}")
+          .setDefaultValue(param.default)
+      }
+
+      // Set the rule as a template if it contains parameters.
+      rule.setTemplate(inspection.params.nonEmpty)
+    }
+
+    // Save the repository.
+    repository.done()
+  }
+}
+
+private[scalastyle] object ScalastyleRulesRepository {
+  private final case class Acc(indent: Boolean, isEmpty: Boolean, text: String)
+
+  final val RepositoryKey = "sonar-scala-scalastyle"
+  final val RepositoryName = "Scalastyle"
+
+  /**
+   * Convert Scalastyle inspection level to SonarQube rule severity.
+   */
+  def levelToSeverity(level: Level): Severity = level match {
+    case InfoLevel    => Severity.INFO
+    case WarningLevel => Severity.MINOR
+    case ErrorLevel   => Severity.MAJOR
+  }
+
+  /**
+   * Convert Scalastyle inspection parameter type to SonarQube rule parameter type.
+   */
+  def parameterTypeToRuleParamType(typ: ParameterType): RuleParamType = typ match {
+    case StringType  => RuleParamType.STRING
+    case IntegerType => RuleParamType.INTEGER
+    case BooleanType => RuleParamType.BOOLEAN
+    // TODO: RuleParamType.TEXT is used for header parameter of the HeaderMatchesChecker inspection - is this necessary?
```

Comment:
The TEXT parameter seems to be used only for the `header` parameter of the `org.scalastyle.file.HeaderMatchesChecker` inspection so that the user gets a bigger input field instead of a small text field to define a text which should match the first lines of each file - I'll need to update this.

http://www.scalastyle.org/rules-dev.html#org_scalastyle_file_HeaderMatchesChecker